### PR TITLE
implemented controller using DTOs on a service layer

### PR DESCRIPTION
**Pull request by annton-shakur**

### CODE REVIEW

File: src/main/java/mate/academy/onlinebookstoreproject/controller/BookController.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/books")
+@RequiredArgsConstructor
+public class BookController {
+    @Autowired
+    private BookService bookService;
```
Comment (DEz-1996) [inline src/main/java/mate/academy/onlinebookstoreproject/controller/BookController.java]: You use @RequiredArgsConstructor so you don't need @Autowired

File: src/main/java/mate/academy/onlinebookstoreproject/repository/impl/BookRepositoryImpl.java [hunk 2]
```diff
@@ -1,57 +1,55 @@
...
+        try (EntityManager entityManager = factory.createEntityManager()) {
+            return entityManager.createQuery("FROM Book", Book.class).getResultList();
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (EntityManager entityManager = factory.createEntityManager()) {
+            Book book = entityManager.find(Book.class, id);
+            return Optional.ofNullable(book);
```
Comment (DEz-1996) [inline src/main/java/mate/academy/onlinebookstoreproject/repository/impl/BookRepositoryImpl.java:54]: Not sure that is a good practice to use try without catch

**Review by DEz-1996 — APPROVED** [review]
Good job! Just some minor points to consider.

File: src/main/java/mate/academy/onlinebookstoreproject/controller/BookController.java (hunk 1 above)
Comment (vovasalyha) [reply src/main/java/mate/academy/onlinebookstoreproject/controller/BookController.java]: Yes, but it's important that field is `final` too.

File: src/main/java/mate/academy/onlinebookstoreproject/repository/impl/BookRepositoryImpl.java (hunk 2 above)
Comment (vovasalyha) [reply src/main/java/mate/academy/onlinebookstoreproject/repository/impl/BookRepositoryImpl.java:54]: If `try` block is used only for resource management then it's ok. But catch block would not be redundant here for sure.

**Review by vovasalyha — APPROVED** [review]
LGTM

**Pull request merged**